### DISCUSSION TITLE
Fade out effect on all Alert-messages.

### DISCUSSION THREAD

**Discussion by pinak3748** [joomla/joomla-cms#35031] [2. Cold Features]
### **We can add fade-out effect on Error, Warning, and all other Alert messages.** 

We have to manually close all the alert messages until we close them they will be there. In my opinion, it will improve the user experience.

**There is already a fade-in effect all and the fade-out effect and in added by some categories like-**

- [ ] Simple fade out after an altered message like password do not match or this field is required, etc.
- [ ] The ones that have a link or which redirect can be as it is and users have to manually do that.
- [ ] One has to fade after another if there is more than one alert message on a page.

**and there can be many cases like this.**

**Comment by pavankumar-tidke** [comment]
we can surely add this as it will improve the functionality as well as user experience of the site.

**Comment by rhythmjain32** [comment]
Nice suggestion though.

**Comment by dgrammatiko** [comment]
There is already a fade in and a fade out effect in the alerts: https://github.com/joomla-projects/custom-elements/blob/1cc751ee7fcf81d6b1e8ce417dee1db35a795a8c/src/scss/alert/alert.scss#L128-L135

**Reply by pinak3748** [reply]
Sorry if I wasn't clear before but I was actually thinking of the auto-fading effect after a certain time span like using functions 
`window.setTimeout` or through javascript EventListeners depending on the situation of alert message like in the above list

**Reply by dgrammatiko** [reply]
That feature is also supported: https://github.com/joomla/joomla-cms/blob/4d992e2245f3e2fdaf2f99d7875089dfe3aec543/build/media_source/system/js/messages.es6.js#L47

**Reply by pinak3748** [reply]
I can't see the effect on my system they remain there until I close them manually 
for eg:
If  I go to  Content>Article>new>and do not enter the title, then the alert message will be there 
```
The form cannot be submitted as it's missing required data.
Please correct the marked fields and try again.
```
I have to close it myself.
The code is there but I am not seeing its implementation

**Reply by dgrammatiko** [reply]
> I can't see the effect on my system they remain there until I close them manually

That's because it's not used at the moment. There were some long discussions about self-destructing the alerts and since there was no consensus the option was left for template/dev creators to implemented. So, in short, although core provides the functionality it doesn't enforce it

**Reply by pinak3748** [reply]
it is just that I thought it's missing it and I have checked for almost all types of alert messages. 
Ok so we have the code but we are not using it as of now.

**Reply by dgrammatiko** [reply]
> Ok so we have the code but we are not using it as of now.

Yes, I wrote the code but I can't force it, it doesn't work that way 🤷‍♂️
